pyquest/drawer.py

```diff
@@ -445,58 +566,91 @@ def draw_gate(gate, column, num_qubits, plt, ax):
     # draw vertical connector lines (at back)
     for line in lines:
 
-        # avoid drawing zero-length lines (eles matplotlib throws)
+        # avoid drawing zero-length lines (else matplotlib throws)
         if line[0] == line[1]:
             continue
 
-        (a,b),(c,d) = line
+        (a, b), (c, d) = line
         plt.plot(
-            (a,c), (b,d), 
-            color=get_vertical_connector_color(gate), 
-            zorder=layer.VERTICAL_CONNECTOR)
-
-    # draw control dots
-    for dot in dots:
-        ax.add_patch(plt.Circle(
-            dot, size.CONTROL_CIRCLE_RADIUS, 
-            color=get_control_qubit_color(gate), 
-            zorder=layer.CONTROL_CIRCLE))
+            (a, c),
+            (b, d),
+            color=colors.get_vertical_connector_color(gate),
+            zorder=layer.VERTICAL_CONNECTOR,
+        )
+
+    # Draw control dots
+    for i, dot in enumerate(dots):
+        control_color = (
+            colors.get_control_qubit_color()
+            if not isinstance(gate, U) or not gate.control_pattern
+            else (
+                colors.get_control_qubit_color()
+                if gate.control_pattern[i] == 1
+                else colors.get_fig_color()
+            )
+        )
+
+        ax.add_patch(
+            plt.Circle(
+                dot,
+                size.CONTROL_CIRCLE_RADIUS,
+                edgecolor=colors.get_gate_edge_color(gate),
+                facecolor=control_color,
+                linewidth=1.5,
+                zorder=layer.CONTROL_CIRCLE,
+            )
+        )
 
     # draw the main body of the gate; possibly labelled rectangles, or bespoke symbols
     draw_gate_body(gate, column, rectangles, plt, ax)
 
 
-def draw_circuit(gates):
-
-    # get matplotlib handles
-    ax = plt.gca()
+def draw_circuit(gates, theme="bw", filename=None):
 
     # determine circuit layout
     gate_columns = get_circuit_columns(gates)
     num_columns = 1 + max(gate_columns)
     num_qubits = get_num_qubits(gates)
 
+    # get matplotlib handles and set the canvas size
+    mpl_figure = plt.figure()
+    mpl_figure.set_size_inches(num_columns, num_qubits)
+    ax = plt.gca()
+
+    # set global color theme
+    global colors
+    colors = Colors(theme)
+
+    # Set the background color
+    mpl_figure.patch.set_facecolor(colors.get_fig_color())
+    ax.set_facecolor(colors.get_fig_color())
+
     # draw horizontal qubit stave
     for q in range(num_qubits):
         plt.plot(
-            [-.5, num_columns+.5], [q+.5, q+.5], 
```

Comment:
you here arbitrarily split these arguments over two lines, throwing away their semantic connection (that they are lists of corresponding `x` and `y` values, and are ergo distinct from the logically uncoupled kwargs that follow)